src/mono/mono.proj

```diff
@@ -31,13 +31,13 @@
     </PropertyGroup>
     <PropertyGroup>
         <_MonoConfigureParams Condition="$(MonoEnableLLVM) == true">$(_MonoConfigureParams) --enable--llvm </_MonoConfigureParams>
-        <_MonoConfigureParams>$(_MonoConfigureParams) --with-core=only CC=clang CXX=clang++ CFLAGS="$(_MonoExtraCFLAGS)" CXXFLAGS="$(_MonoExtraCXXFLAGS)"</_MonoConfigureParams>
+        <_MonoConfigureParams>$(_MonoConfigureParams) --enable-maintainer-mode --enable-compile-warnings --with-core=only CFLAGS="$(_MonoExtraCFLAGS)" CXXFLAGS="$(_MonoExtraCXXFLAGS)"</_MonoConfigureParams>
     </PropertyGroup>
 
     <Message Text="Configuring Mono with '$(_MonoConfigureParams)'" Importance="High" />
     <MakeDir Directories="$(MonoObjDir)" />
     <Exec Command="NOCONFIGURE=1 $(MonoProjectRoot)autogen.sh" />
-    <Exec Command="$(MonoProjectRoot)configure $(_MonoConfigureParams)" WorkingDirectory="$(MonoObjDir)" />
+    <Exec Command="bash -c 'source $(RepositoryEngineeringDir)native/init-compiler.sh $(Platform) clang &amp;&amp; $(MonoProjectRoot)configure $(_MonoConfigureParams)'" WorkingDirectory="$(MonoObjDir)" />
```

Comment:
@akoeplinger, would it be possible to let `eng/build.sh` determine which compiler to use? i.e. the hardcoded `clang` to be replaced by `$(Compiler)` defined here: https://github.com/dotnet/runtime/blob/2ef008e29b7f52d6dee8be3ea0bb675651cccb8f/eng/build.sh#L275-L292
Doing that will help systems which require gcc build (`runtime/build.sh -gcc ...`).

Reply:
@am11 sure, that would be great!